libs/api-bundle/tests/Security/ManageApiToken/ManageApiTokenAuthenticatorTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ApiBundle\Tests\Security\ManageApiToken;
+
+use Generator;
+use Keboola\ApiBundle\Attribute\ManageApiTokenAuth;
+use Keboola\ApiBundle\Security\ManageApiToken\ManageApiClientFactory;
+use Keboola\ApiBundle\Security\ManageApiToken\ManageApiToken;
+use Keboola\ApiBundle\Security\ManageApiToken\ManageApiTokenAuthenticator;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+class ManageApiTokenAuthenticatorTest extends TestCase
+{
+    /**
+     * @dataProvider provideAuthorizeTokenSuccessData
+     */
+    public function testAuthorizeTokenSuccess(ManageApiTokenAuth $authAttribute, ManageApiToken $token): void
+    {
+
+        $authenticator = new ManageApiTokenAuthenticator(
+            $this->createMock(ManageApiClientFactory::class),
+        );
+
+        $authenticator->authorizeToken($authAttribute, $token);
+
+        $this->expectNotToPerformAssertions();
+    }
+
+    /**
+     * @dataProvider provideAuthorizeTokenExceptionsData
+     */
+    public function testAuthorizeTokenExceptions(
+        ManageApiTokenAuth $authAttribute,
+        ManageApiToken $token,
+        string $expectedMessage,
+    ): void {
+        $authenticator = new ManageApiTokenAuthenticator(
+            $this->createMock(ManageApiClientFactory::class),
+        );
+
+        $this->expectException(AccessDeniedException::class);
+        $this->expectExceptionMessage($expectedMessage);
+
+        $authenticator->authorizeToken($authAttribute, $token);
+    }
+
+    public function provideAuthorizeTokenSuccessData(): Generator
+    {
+        yield 'scope needed and provided' => [
+            new ManageApiTokenAuth(['some:scope']),
+            ManageApiToken::fromVerifyResponse([
+                'id' => 123,
+                'description' => 'some-description',
+                'scopes' => ['some:scope'],
+            ]),
+        ];
+
+        yield 'super admin token needed and provided' => [
+            new ManageApiTokenAuth([], true),
```

Comment:
pri tom vytvareni attributu bych pouzil named arguments.. technicky to neni rozdil, ale je to citelnejsi (`new ManageApiTokenAuth(isSuperAdmin: true)`)